pkg/kapis/openpitrix/v1/register.go

```diff
@@ -21,34 +21,36 @@ import (
 	"github.com/emicklei/go-restful"
 	"github.com/emicklei/go-restful-openapi"
 	"k8s.io/apimachinery/pkg/runtime/schema"
-	"kubesphere.io/kubesphere/pkg/apiserver/openpitrix"
 	"kubesphere.io/kubesphere/pkg/apiserver/runtime"
 	"kubesphere.io/kubesphere/pkg/constants"
 	"kubesphere.io/kubesphere/pkg/models"
-	"kubesphere.io/kubesphere/pkg/models/openpitrix/application"
-	"kubesphere.io/kubesphere/pkg/models/openpitrix/type"
+	openpitrix2 "kubesphere.io/kubesphere/pkg/models/openpitrix"
 	"kubesphere.io/kubesphere/pkg/server/errors"
 	"kubesphere.io/kubesphere/pkg/server/params"
+	"kubesphere.io/kubesphere/pkg/simple/client/k8s"
+	op "kubesphere.io/kubesphere/pkg/simple/client/openpitrix"
 	"net/http"
 )
 
-const GroupName = "openpitrix.io"
+const (
+	GroupName = "openpitrix.io"
+	ok        = "ok"
+)
 
 var GroupVersion = schema.GroupVersion{Group: GroupName, Version: "v1"}
 
-var (
-	WebServiceBuilder = runtime.NewContainerBuilder(addWebService)
-	AddToContainer    = WebServiceBuilder.AddToContainer
-)
+func AddToContainer(c *restful.Container, k8s k8s.Client, op op.Client) error {
 
-func addWebService(c *restful.Container) error {
+	if k8s == nil || op == nil {
```

Comment:
since k8s and op is required for openpitrix, no need to nullability here, dependencies is guaranteed to not be null